@YaoJiayi regarding GPU mapped memory, there seems to be mechanisms in NVIDIA libraries that prevent dealing with it directly, even though there is mapping for it in the process. For example, attempting to use the base pointer of the `CuFileMemoryAllocator` tensor, with a small test program:

```
device_str = "cuda:0"
size = 256 * 1024 * 1024
memory_allocator = CuFileMemoryAllocator(size, device=device_str)
cufile_base_pointer = memory_allocator.base_pointer
mv = memoryview((ctypes.c_uint8 * size).from_address(cufile_base_pointer))
with open('/usr/bin/docker', 'rb') as f:
    return f.readinto(mv)
```

Result:

```
Traceback (most recent call last):
  File "test.py", line 128, in <module>
    main()
  File "test.py", line 97, in main
    return f.readinto(mv)
           ^^^^^^^^^^^^^^
OSError: [Errno 14] Bad address
```

This is because that mapping is inaccessible to regular access (see: `/proc/<pid>/maps`):

```
7f1584000000-7f15ac400000 ---p 00000000 00:00 0
```

The `read` system call returns EFAULT because that memory area has no write permissions. I am not sure what the `cudart` library does behind the scenes, but it probably involves translating addresses and making sure memory is pinned.

BTW, it is unrelated to whether `cuFileBufRegister` did anything or not (and for `use_cufile == false` cases, a further change would be to use `GPUMemoryAllocator` instead, forgoing the cufile library entirely).

--- a/lmcache/v1/storage_backend/gds_backend.py
+++ b/lmcache/v1/storage_backend/gds_backend.py
@@ -199,14 +199,16 @@ class GdsBackend(StorageBackendInterface):
         if config.extra_config is not None:
             use_cufile = config.extra_config.get("use_cufile", None)
             if use_cufile is not None:
-                if use_cufile in [False, True]:
-                    logger.info("Getting use_cufile from config")
+                logger.info("Getting use_cufile from config")
+                if isinstance(use_cufile, str):
+                    self.use_cufile = use_cufile.lower() == "true"
+                elif use_cufile in [False, True]:
                     self.use_cufile = use_cufile
-                    use_cufile_from_config = True
                 else:
                     raise RuntimeError(
                         f"Invalid value `{use_cufile}` for use_cufile in extra_config"
                     )
+                use_cufile_from_config = True
 
         if self.fstype in ["tmpfs", "overlayfs"]:
             # TODO: we can replace the auto-detection of unsupported cufile

